cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1833,17 +1833,23 @@ auto convert_table_to_parquet_data(table_input_metadata& table_meta,
   // Initialize data pointers in batch
   uint32_t const num_stats_bfr =
     (stats_granularity != statistics_freq::STATISTICS_NONE) ? num_pages + num_chunks : 0;
-  rmm::device_buffer uncomp_bfr(max_uncomp_bfr_size, stream);
-  rmm::device_buffer comp_bfr(max_comp_bfr_size, stream);
-  rmm::device_buffer col_idx_bfr(column_index_bfr_size, stream);
+  std::vector<rmm::device_buffer> uncomp_bfr;
+  std::vector<rmm::device_buffer> comp_bfr;
+  std::vector<rmm::device_buffer> col_idx_bfr;
+  for (std::size_t i = 0; i < batch_list.size(); ++i) {
+    uncomp_bfr.emplace_back(max_uncomp_bfr_size, stream);
```

Comment:
This can lead to a large overuse of memory if there are unbalanced batches. For instance, in testing I had a file that did one batch of 14 row groups, and a second batch of a single row group, all sized the same. For that case this will have two buffers large enough for 14 rowgroups, 28 total, instead of the 15 needed.  Could you add a note here to that effect?

Reply:
It seems that the alternative solution (https://github.com/rapidsai/cudf/pull/13438#issuecomment-1561999150) should be better. With that, we won't have overuse memory.